File: Leetcode/maximum-depth-of-binary-tree.js
//https://leetcode.com/problems/maximum-depth-of-binary-tree/

var maxDepth = function(root) {
  if (!root) return 0;
  let leftHeight = maxDepth(root.left);
  console.log(`left height from ${root.val}: ${leftHeight}`);
  let rightHeight = maxDepth(root.right);
  console.log(`right height from ${root.val}: ${rightHeight}`)
  console.log(Math.max(leftHeight, rightHeight) + 1);
  return (Math.max(leftHeight, rightHeight) + 1);
};

File: Leetcode/Maximum_Depth_Of_Binary_Tree.py
class Solution:
    def maxDepth(self,root: Optional[TreeNode])->int:
        if root==None:
            return 0
        return Solution.dfs(root,0)

    def dfs(root,cnt):
        cnt+=1
        l,r=0,0
        if root.left==None and root.right==None:
            return cnt
        if root.left:
            l=Solution.dfs(root.left,cnt)
        if root.right:
            r=Solution.dfs(root.right,cnt)
        return max(l,r)
        